api/views.py

```diff
@@ -286,28 +282,36 @@ def aggregate_shift_content(self, shifts):
         for date in dates:
             shifts_of_date = shifts.filter(started__date=date)
 
-            worked_shifts = shifts_of_date.filter(type="st")
-            vacation_or_sick_shifts = shifts_of_date.exclude(type="st")
-            # calculate time worked
-            worked_time = worked_shifts.aggregate(
+            worktime = shifts_of_date.aggregate(
                 work_time=Coalesce(
                     Sum(F("stopped") - F("started"), output_field=DurationField()),
                     datetime.timedelta(0),
                 )
             )["work_time"]
-            # calculate time not present
-            sick_or_vacation_time = vacation_or_sick_shifts.aggregate(
-                sick_or_vac_time=Coalesce(
-                    Sum(F("stopped") - F("started"), output_field=DurationField()),
-                    datetime.timedelta(0),
-                )
-            )["sick_or_vac_time"]
 
-            vacation_or_sick_type = ""
-            if vacation_or_sick_shifts.exists():
-                vacation_or_sick_type = (
-                    vacation_or_sick_shifts.first().get_type_display()
-                )
+            breaktime = calculate_break(
+                shifts_of_date,
+            )
+
+            if datetime.timedelta(hours=6) < worktime <= datetime.timedelta(hours=9):
```

Comment:
Das Doppelt sich jetzt in der Validation vom ContractSerializer.

Reply:
evtl. besser ein neues file erstellen was wir `validators.py` nennen.

Reply:
das ist im Fall des Stundenzettels doch aber kein wirklicher Validator